app/src/features/surveys/components/EditDeleteStubCard.tsx

```diff
@@ -0,0 +1,68 @@
+import { mdiPencilOutline, mdiTrashCanOutline } from '@mdi/js';
+import Icon from '@mdi/react';
+import { Card, CardHeader, IconButton } from '@mui/material';
+import { grey } from '@mui/material/colors';
+import React from 'react';
+
+export type ISubHeaderData = Record<string, string | number | undefined> | string;
```

Comment:
Nit: this type is not an interface, so you can just call this `SubHeaderData`.

I might reconsider accepting an `object | string`-like React prop, since the resulting UI behvaiour varies quite a bit. Two things to consider:

1. Use `type SubHeaderData = { key: string; value?: string | number }`. This way, if I want my header to just display a single string, I can pass `header={{ key: 'My Header }}`.
2. Inline your header rendering method:
```javascript
<CardHeader
  subheader={
    props.header.map((header) => [header.key, header.value].filter(Boolean).join(' • ')).join(' | ')
  }
  ...
/>

Reply:
If you find this strategy compelling, you may reconsider the property names for `SubHeaderData`, e.g. `title` instead of `key`

Reply:
Fixed the type.

It might almost be beneficial to only accept strings for the subHeaderData. 
Allow the subheader format to be handled by the parent component for more flexibility.

Just from exploring this solution, the objects passed to subHeaderData become very verbose.